exporter/metric/test/test.go

```diff
@@ -42,25 +53,34 @@ func createNumber(desc *export.Descriptor, v float64) core.Number {
 func (p *CheckpointSet) AddGauge(desc *export.Descriptor, v float64, labels ...core.KeyValue) {
 	ctx := context.Background()
 	gagg := gauge.New()
+	agg := p.Add(desc, gagg, labels...)
 	_ = gagg.Update(ctx, createNumber(desc, v), desc)
 	gagg.Checkpoint(ctx, desc)
-	p.Add(desc, gagg, labels...)
+	if agg != gagg {
```

Comment:
Can you add a comment about this new test support?

Reply:
Sure, this was a little bit enigmatic. I've refactored this a bit and added some docs.